src/pkg/leaderElection.go

```diff
@@ -43,8 +44,26 @@ func RunLeaderElection(client *clientset.Clientset, lockName, lockIdentity, lock
 				log.Info().Msgf("Perform Migration")
 				for {
 					log.Info().Msgf("leader is %s", lockIdentity)
-					log.Info().Msgf("Getting replica count...")
-					time.Sleep(5 * time.Second)
+					replicaCount, err := getReplicas()
+					if err != nil {
+						log.Fatal().Msgf("Failed to get replica count")
+					}
+					log.Info().Msgf("Setting replica count to %v", replicaCount)
+					retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+						deploymentsClient := client.AppsV1().Deployments(lockNamespace)
+						result, getErr := deploymentsClient.Get(context.TODO(), lockName, metav1.GetOptions{})
+						if getErr != nil {
+							log.Fatal().Msgf("Failed to get latest version of Deployment: %v", getErr)
```

Comment:
should return the error here since the function signature this is in expects and error to be return - which means you can early out if needed.

Reply:
Oh yep, I guess I expected the Fatal() part to exit the function but I can make that change.

Reply:
oh i didn't even see that - but ya you don't want to fatal here as that will cause panic's - you need to log the message and return the error.